vmdb/app/controllers/application_controller/performance.rb

```diff
@@ -176,7 +176,11 @@ def perf_refresh_data
   def perf_set_or_fix_dates(options)
     # Get start/end dates in selected timezone
     tz = options[:time_profile_tz] || options[:tz]  # Use time profile tz or chosen tz, if no profile tz
-    s, e = @perf_record.first_and_last_capture
+    s, e = @perf_record.first_and_last_capture('hourly')
+    if s.nil?
+      s, e = @perf_record.first_and_last_capture('realtime')
+      options[:typ] = "realtime"
```

Comment:
@h-kataria even though I made this change as you suggested I strongly advise against it. In fact I'd rather revert to the previous patch (without `options[:typ] = "realtime"`).

The code in this file is already a little hard to follow, changing the type here is something that would be really hard to find for anyone else that is going to debug an issue in the future.

Also consider that by default the chart is "Daily" which means that currently when you pass `first_and_last_capture('hourly')` it still doesn't mean that you have the daily metrics.
So my change here for "realtime" is a little irrelevant.